### DISCUSSION TITLE
RustDesk server pro connection issues

### DISCUSSION THREAD

**Discussion by typecek192837** [rustdesk/rustdesk-server-pro#372] [Q&A]
hello, i would like to ask why my connection isnt stable, i have self-hosting server and when im trying to connect to pc with rustdesk client instalation either it connects or shows that the keys don't match or that the remote desktop is offline but ussualy it just doesnt connect,i have been testing it on local site on another pc that we have here,  i have the same credentials on both of pcs and it just doesnt work, and it shows that client is offline even if isnt, pls help.
And please simplify it im not that experienced, thank you.
![i1](https://github.com/user-attachments/assets/10490889-0ebf-4d0e-96cd-2a5c73900f04)
![i2](https://github.com/user-attachments/assets/20a14d05-1811-4aa5-8c3b-4f187ea2ca03)
![i3](https://github.com/user-attachments/assets/06466c55-364f-4ab7-b7df-c929a4e2c711)
![i4](https://github.com/user-attachments/assets/b36fdc71-9c9a-43a1-b104-bf073169044e)

**Comment by rustdesk** [comment]
Following this.

- copy the config to clipboard
![image](https://github.com/user-attachments/assets/1b907e59-753f-4ada-9475-f0b9a69f1593)

- import the config from clipboard
![image](https://github.com/user-attachments/assets/495a1a2c-3103-4487-9a68-c706e68dfcd7)

**Reply by typecek192837** [reply]
now its just "connecting" and then deadline cuts it, then it reconnects and this happens over and over again
![image](https://github.com/user-attachments/assets/46d60691-3cae-42fc-bf8c-f98be3afe08f)
![image](https://github.com/user-attachments/assets/9ced8ece-3e6a-409e-99b3-8868aca851fb)

**Reply by rustdesk** [reply]
Try this https://github.com/rustdesk/rustdesk/wiki/FAQ#force-relay

**Reply by typecek192837** [reply]
i figured it out, i didn't installed hbbr and hbbs to the server c:, thanks

**Reply by typecek192837** [reply]
but now the web console wont open, did i do something wrong?

**Reply by rustdesk** [reply]
it seems `21114` port blocked by your firewall, or you did something wrong I do not know.

**Reply by typecek192837** [reply]
it isnt blocked

**Reply by hewm** [reply]
wget rustdesk.com/pro.yml -O compose.yml
docker compose up -d

```
CONTAINER ID   IMAGE                                 COMMAND                  CREATED         STATUS                   PORTS                                         NAMES
8ce3d98de2b0   rustdesk/rustdesk-server-pro:latest   "hbbs"                   9 minutes ago   Up 9 minutes                                                           hbbs
17368597f475   rustdesk/rustdesk-server-pro:latest   "hbbr"                   9 minutes ago   Up 9 minutes                                                           hbbr
```


Is 21114 TCP or UDP?
source： 0.0.0.0/0
TCP: 21114 21115 21116 21117 21118 21119
UDP: 21116

http://xx.xxx.xxx.xxx:21114
This page is not working properly
